File: designPattern/template/java/StringDisplay.java

public class StringDisplay extends AbstractPrint {
	String name;
	int width;

	public StringDisplay(String name) {
		this.name = name;
		this.width = name.length();
	}

	public void open() {
		this.printLine();
	}

	public void print() {
		System.out.println("|" + this.name + "|");
	}

	public void close() {
		this.printLine();
	}

	public void printLine() {
		StringBuilder line = new StringBuilder();
		line.append("*");
		for (int i = 0; i < this.width; i++)
			line.append("-");

		line.append("*");
		System.out.println(line);
	}
}

File: designPattern/template/python/StringDisplay.py
from AbstractPrint import AbstractPrint


class StringDisplay(AbstractPrint):

    def __init__(self, input_string):
        self.output_string = input_string
        self.width = len(input_string)
        self.result = ""

    def open(self):
        self.printLine()

    def print_input(self):
        print("|" + self.output_string + "|")

    def close(self):
        self.result = ""
        self.printLine()

    def printLine(self):
        self.result += "+"
        for i in range(0, self.width):
            self.result += "-"
        self.result += "+"
        print(self.result, end="\n")


if __name__ == "__main__":
    hi = StringDisplay("hi")
    hi.display()
